analytics/consumer.py

```diff
@@ -71,14 +79,23 @@ def next(self):
         items = []
 
         start_time = monotonic.monotonic()
+        total_size = 0
 
         while len(items) < self.upload_size:
             elapsed = monotonic.monotonic() - start_time
             if elapsed >= self.upload_interval:
                 break
             try:
                 item = queue.get(block=True, timeout=self.upload_interval - elapsed)
-                items.append(item)
+                item_size = len(json.dumps(item, cls=DatetimeSerializer).encode())
+                if item_size > MAX_MSG_SIZE:
+                    self.log.error('Item exceeds 32kb limit, dropping. (%s)', str(item))
```

Comment:
To avoid lots of nesting, I like to use early returns/breaks.

```
if item_size > MAX_MSG_SIZE:
   ....
   break

// Else statement goes here, without the extra nesting
items.append(item)
...
```